features/graph/api/src/main/java/org/opennms/netmgt/graph/api/generic/GenericGraphContainer.java

```diff
@@ -39,6 +39,7 @@
 import org.opennms.netmgt.graph.api.info.GraphInfo;
 
 // TODO MVR we must rework the generic and simple objects a bit more... that is kinda weird how it is implemented. The graph could work as an example but is also not fully defined yet
+// TODO: Patrick discuss with mvr is we should make the GraphContaner partially immutable similar to GenericGraph?
```

Comment:
This might be another candidate for adding a `immutableCopy()` method.

Reply:
Each `Generic*` element should have an immutable counterpart.